pkg/scheduler/plugins/elasticquota/plugin.go

```diff
@@ -0,0 +1,560 @@
+		return
+	}
+	podReq, _ := resource.PodRequestsAndLimits(pod)
+	g.groupQuotaManager.UpdateGroupDeltaRequest(quotaName, podReq)
+	if pod.Status.Phase == corev1.PodRunning {
+		g.groupQuotaManager.UpdateGroupDeltaUsed(quotaName, podReq)
+		g.quotaOverUsedRevokeController.addPodWithQuotaName(quotaName, pod)
+	}
+	klog.V(5).Infof("OnPodAddFunc %v.%v add request success, quotaName:%v", pod.Namespace, pod.Name, quotaName)
+}
+
+func (g *Plugin) OnPodUpdate(oldObj, newObj interface{}) {
+	oldPod := oldObj.(*corev1.Pod)
+	newPod := newObj.(*corev1.Pod)
+
+	if oldPod.ResourceVersion == newPod.ResourceVersion {
+		klog.Warningf("update pod warning, update version for the same:%v", newPod.Name)
+		return
+	}
+
+	oldQuotaName := g.getPodAssociateQuotaName(oldPod)
+	newQuotaName := g.getPodAssociateQuotaName(newPod)
+	g.Lock()
+	defer g.Unlock()
+
+	if err := g.elasticQuotaInfos[oldQuotaName].deletePodIfPresent(oldPod); err != nil {
+		return
+	}
+	if err := g.elasticQuotaInfos[newQuotaName].addPodIfNotPresent(newPod); err != nil {
+		return
+	}
+	// delete oldPod's Request
+	podReq, _ := resource.PodRequestsAndLimits(oldPod)
+	podReqTmp := quotav1.Subtract(corev1.ResourceList{}, podReq)
+	g.groupQuotaManager.UpdateGroupDeltaRequest(oldQuotaName, podReqTmp)
+
+	// add newPod's Request
+	podReq, _ = resource.PodRequestsAndLimits(newPod)
+	g.groupQuotaManager.UpdateGroupDeltaRequest(newQuotaName, podReq)
+
+	if oldQuotaName != newQuotaName {
+		if oldPod.Status.Phase == corev1.PodRunning {
+			g.groupQuotaManager.UpdateGroupDeltaUsed(oldQuotaName, podReqTmp)
+			g.quotaOverUsedRevokeController.erasePodWithQuotaName(oldQuotaName, oldPod)
+		}
+		if newPod.Status.Phase == corev1.PodRunning {
+			g.groupQuotaManager.UpdateGroupDeltaUsed(newQuotaName, podReq)
+			g.quotaOverUsedRevokeController.addPodWithQuotaName(newQuotaName, newPod)
+		}
+	}
+
+	klog.V(5).Infof("OnPodUpdateFunc %v.%v add request success, quotaName:%v", newPod.Namespace, newPod.Name, newQuotaName)
+}
+
+func (g *Plugin) OnPodDelete(obj interface{}) {
+	pod, ok := obj.(*corev1.Pod)
+	if !ok {
+		return
+	}
+
+	g.Lock()
+	defer g.Unlock()
+
+	quotaName := g.getPodAssociateQuotaName(pod)
+	if err := g.elasticQuotaInfos[quotaName].deletePodIfPresent(pod); err != nil {
+		// the pod is not added to the g.elasticQuotaInfos[quotaName], so no need to call UpdateGroupDeltaUsed.
+		return
+	}
+	podReq, _ := resource.PodRequestsAndLimits(pod)
+	podReqTmp := quotav1.Subtract(corev1.ResourceList{}, podReq)
+	g.groupQuotaManager.UpdateGroupDeltaRequest(quotaName, podReqTmp)
+
+	g.groupQuotaManager.UpdateGroupDeltaUsed(quotaName, podReqTmp)
+	g.quotaOverUsedRevokeController.erasePodWithQuotaName(quotaName, pod)
+	klog.V(5).Infof("OnPodDeleteFunc %v.%v delete request success", pod.Namespace, pod.Name)
+}
+
+func (g *Plugin) OnNodeAdd(obj interface{}) {
+	node, ok := obj.(*corev1.Node)
+	if !ok {
+		return
+	}
+
+	klog.V(5).Infof("OnNodeAddFunc add node[%v]", node.Name)
```

Comment:
this log should like update\delete position and content

Reply:
fixed